preset-prettier integration broken after update of remark-preset-prettier to 2.0.0

### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

eslint-plugin-mdx 2.0.4

### Link to runnable example

Not runnable, but easily made runnable: https://github.com/enqueue/eslint-mdx-reproducer

### Steps to reproduce

1. Set up a project with eslint-plugin-mdx and remark-preset-prettier version 2.0.0
2. Add "preset-prettier" plugin to projects `remarkrc`
3. Run `eslint .`

(Hopefully) minimal reproducible example over here: https://github.com/enqueue/eslint-mdx-reproducer

Node version is 18.9

### Expected behavior

MDX and MD files are linted according to rules.

### Actual behavior

Error message `0:0  error  Parsing error: Cannot find module 'preset-prettier'`.

### Runtime

Other (please specify in steps to reproduce)

### Package manager

npm v8

### OS

Linux

### Build and bundle tools

_No response_

### Workaround

My current workaround is to pin the dependency of remark-preset-prettier to 1.0.2

--- a/.changeset/proud-trainers-sort.md
+++ b/.changeset/proud-trainers-sort.md
@@ -1,0 +1,5 @@
+---
+"eslint-mdx": patch
+---
+
+fix: handle pure ESM robustly

--- a/packages/eslint-mdx/src/helpers.ts
+++ b/packages/eslint-mdx/src/helpers.ts
@@ -90,8 +90,14 @@ export const loadModule = async <T>(modulePath: string): Promise<T> => {
         // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-unsafe-return
         return require(modulePath)
       } catch (err) {
+        const code = (err as { code: string }).code
         /* istanbul ignore if */
-        if ((err as { code: string }).code === 'ERR_REQUIRE_ESM') {
+        if (
+          code === 'ERR_REQUIRE_ESM' ||
+          // A pure ESM could have no `exports.require` and then throw the following error,
+          // related to #427.
+          code === 'ERR_PACKAGE_PATH_NOT_EXPORTED'
+        ) {
           // Load the ESM configuration file using the TypeScript dynamic import workaround.
           // Once TypeScript provides support for keeping the dynamic import this workaround can be
           // changed to a direct dynamic import.

